examples/vercel-ai-chatbot/lib/hooks/use-vellum-chat.tsx

```diff
@@ -67,85 +142,22 @@ const useVellumChat = ({
             }
             throw e
           })
-          if (!resultChunk?.value) {
+          if (resultChunk?.done || !resultChunk) {
             break
           }
 
-          const decoder = new TextDecoder('utf-8')
-          const resultChunkValueString = decoder.decode(resultChunk.value)
-          const cleanedChunkValueString = resultChunkValueString
-            .split(/[\n\r]+/)
-            .map(s => s.trim())
-            .filter(s => s !== '')
-            .join(', ')
-          const parsedChunkValueArray = JSON.parse(
-            `[${cleanedChunkValueString}]`
-          ) as WorkflowResultEventOutputData[]
-
-          parsedChunkValueArray.forEach(parsedChunkValue => {
-            if (
-              parsedChunkValue.state === 'REJECTED' &&
-              parsedChunkValue.type === 'ERROR'
-            ) {
-              throw new Error(parsedChunkValue.value?.message)
+          buffer = concatBuffers(buffer, resultChunk.value)
+          while (buffer.length >= 4) {
```

Comment:
[nit] Another comment here for why four and about the buffer logic would be helpful

Reply:
I think pulling the 4 out into a constant e.g. `UNIT32_SIZE` would make this more clear